### ISSUE TITLE
[jruby 1.7.21] ManyVarsDynamicScope.java:71:in `getValue': java.lang.ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by atambo** [jruby/jruby#3259]
We are seeing this error show up every once and a while when running our tests on our CI system:

```
ManyVarsDynamicScope.java:71:in `getValue': java.lang.ArrayIndexOutOfBoundsException
15:25:06    from DVarNode.java:118:in `interpret'
15:25:06    from CallOneArgNode.java:57:in `interpret'
15:25:06    from IfNode.java:110:in `interpret'
15:25:06    from NewlineNode.java:105:in `interpret'
15:25:06    from BlockNode.java:71:in `interpret'
15:25:06    from ASTInterpreter.java:112:in `INTERPRET_BLOCK'
15:25:06    from Interpreted19Block.java:206:in `evalBlockBody'
15:25:06    from Interpreted19Block.java:194:in `yield'
15:25:06    from Interpreted19Block.java:125:in `call'
15:25:06    from Block.java:101:in `call'
15:25:06    from RubyProc.java:290:in `call'
15:25:06    from RubyProc.java:228:in `call'
15:25:06    from Ruby.java:3183:in `tearDown'
15:25:06    from Ruby.java:3167:in `tearDown'
15:25:06    from Main.java:294:in `internalRun'
15:25:06    from Main.java:217:in `run'
15:25:06    from Main.java:197:in `main'
```

**Comment by enebo** [comment]
@atambo wow this looks really familar....it would seem tearDown is running Ruby code and what it is depending is gone perhaps.  I do not remember why this was happening before...

**Comment by headius** [comment]
@atambo It may be possible to get a better trace by specifying -X+C when you run; that should cause the code being executed above to get compiled at load.

**Comment by headius** [comment]
@atambo Any more information on this?

**Comment by atambo** [comment]
I added ```-X+C``` to ```JRUBY_OPTS```. I'll attach the new trace whenever it happens next.

**Comment by atambo** [comment]
Looks like I get the same trace even with the ```-X+C``` added to ```JRUBY_OPTS``` :-(

**Comment by headius** [comment]
This is very old and there have been various fixes to 1.7 since then. In addition, 1.7.27 will be the last of the 1.7 line and we strongly encourage all users to migrate to JRuby 9k. Closing.
